src/kaleidoscope/MatrixAddr.h

```diff
@@ -37,6 +37,8 @@ class MatrixAddr {
 
   static constexpr uint8_t invalid_state = 255;
 
+  static constexpr ThisType invalid{invalid_state};
```

Comment:
I'd prefer that we default to using = assignment where possible

Reply:
<pedantic>This isn't technically _assignment_; it's just the "normal" C++ way to call a constructor for initialization, and because of the "most vexing parse" issue, the curly braces are better to use than parens.</pedantic>

That said, we can use "=" here, and I'm okay with changing it if that's what you prefer, but it does look more confusing to me:
```c++
static constexpr ThisType invalid = ThisType(invalid_state);
```
The way I wrote it is much more idiomatic C++, and I personally prefer it, but this certainly isn't a hill I'll fight and die on.